### PR TITLE
Interactive plots

### PR DESCRIPTION
**Pull request by isaac-florence**
First draft but imagine few edits to be made.

**Very** simple but gives as much wisdom as will be required for almost all users. I can expand with more examples if you think it's necessary, Neale?

When knitting I noticed that {klippy} doesn't appear to be available for R 4.0.3 ? This raises the wider issue of package and R version control for this project. Is it worth us using {renv} for this project? Also a good Advanced section topic to include?

### CODE REVIEW

File: pages/interactive_plots.Rmd [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+Data visualisation is increasingly required to be interrogable by the audience. Consequently creating interactive plots are becoming common. There are several ways to include these but the two most important are {*plotly*} and {*shiny*}. 
+
+{**Shiny**} is covered in another part of this handbook, so we will only cover {**plotly**} here.
+#TODO - link to shiny page
+
+
+<!-- ======================================================= -->
+## Overview {.tabset .tabset-fade .tabset-pills}
+
+Making plots interactive is a much more daunting prospect than it transpires to be, thanks to some fantastic tool. However, like much of R, if you want to really customise something, you can make it much harder, often for a good payoff
```
Comment (nsbatra) [inline pages/interactive_plots.Rmd:40]: "than it transpires to be" - just be careful that language is accessible to someone with more basic english language skills

File: pages/interactive_plots.Rmd [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+```
+  
+Count for plotting
+```{r manipulate_show2, eval=F}
+## Find number of cases in each week by their outcome
+linelist <- linelist %>% 
+  dplyr::count(week_earliest, outcome)
+```
+  
+Make into a plot
```
Comment (nsbatra) [inline pages/interactive_plots.Rmd:106]: Consider moving the plotting steps here (and below) out of the Preparation tab and into another tab ("Plot", or something). In the other handbook pages, Preparation is generally the import/clean/aggregate steps, but not the actual plotting steps.

**Review by nsbatra — DISMISSED** [review]
Thanks Issac! Looking great. 
I left a few small suggestions and we can review in more depth later. Feel free to merge as you wish. 
Yes, I think a couple other examples might be good but then again we may need to watch the file size. If we add one or two others, maybe a GIS example of coordinates? or another different plot type on here https://plotly.com/ggplot2/
Regarding the renv and klippy - good catch! How would we integrate renv? Im not so familiar. I'll investigate too...
Thanks again.

**Comment by isaac-florence** [comment]
1-Very good point RE English style!
2-Also very wise, done here now

RE further additions, now added a heat map and may add a line chart of some sort too later. RE mapping, it turns out plotly & geom_sf in Rmd are not friends. Works well in RStudio viewer as expected but in knitter html it is so far zoomed out that you can't see anything. Oddly setting the zoom requires changing the JavaScript which doesn't seem sensible. Easiest option for open source reproducibility is fiddling about with GeoJSON object using a basic custom function and {purrr}. Was worth it but hoping there is a simpler way to do this.

**Review by nsbatra — APPROVED** [review]
Thanks for this update @isaac-florence ! 
Approving because I want to see how this looks in the html. Am excited to see... 
I think the complications you mentioned of zoom/GeoJSON/Rmarkdown etc are great to address, but don't need to be the 'main course' for the GIS mapping example. Most users will start by creating these outputs in RStudio only, so let's offer them a first code example with that use-case in mind, if that is more simple. You can always use eval=F on the chunk and print a pre-saved image below to show the output. Then explain the troubles with Rmarkdown and offer a solution for that usecase. 
What do you think of that way forward?

**Comment by nsbatra** [comment]
Oh one other thing. Noticed that theres a column called "sovereignt" - is there a reason there's no "y" on the end?

**Comment by isaac-florence** [comment]
There's a knitted html as a one off in the outputs subfolder if you want to see it there!
That's a good point, however I think after playing with that although you can have it zoomed correctly the interactivity/hovering doesn't work well with geom_sf either... I will add it in for completeness but I think it's a good "common pitfall" example rather than the baseline!

Regarding "soverignt" bizarrely that is just what the column is called when you download it, and that is the best website for global OS geoJSON files from what I can see so wanted to make it as simple to reproduce as possible

**Pull request merged**